Updated Documentation and implemented feedback (#286)
Azure/arc_jumpstart_docs: docs/azure_arc_jumpstart/azure_edge_iot_ops/aks_edge_essentials_single_esa/_index.md

---
type: docs

> title: "Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment"

Reviewer: Discover should be removed from the title.

> linkTitle: "Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment"

Reviewer: Discover should be removed.

The scenario should just be titled "Edge Storage Accelerator on AKS Edge Essential single node deployment"

Reviewer: Discover is the start of the Akri scenario where you actually discover cameras with Akri so its just weird to see it reused here in a different way

weight: 8
description: >
---

> ## Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment 

Reviewer: ```suggestion
## Edge Storage Accelerator on AKS Edge Essentials single node deployment 
```

In this scenario, a computer vision AI model detects defects in bolts by analyzing video from a supply line video feed streamed over RTSP. The identified defects are then stored in a container within a storage account using ESA.

The automation in this scenario deploys Edge Storage Accelerator on an AKS Edge Essentials single-node deployment running in an Azure virtual machine. The provided ARM template creates the Azure resources and configures the LogonScript.ps1 custom script extension which handles AKS Edge Essentials cluster creation, Azure Arc onboarding for the Azure VM and AKS Edge Essentials cluster, and Edge Storage Accelerator deployment. Once AKS Edge Essentials is deployed [Edge Storage Accelerator](https://learn.microsoft.com/azure/azure-arc/edge-storage-accelerator/overview) is installed as a Kubernetes service that exposes a CSI driven storage class for use by applications in the Edge Essentials Kubernetes cluster.

> ![Screenshot ARM template output](./01.png)

Reviewer: ```suggestion
![Screenshot showing scenario architecture diagram](./01.png)
```

Reviewer: I suggest moving this screenshot to earlier in the document, ideally after the second paragraph.

    For example:

    ```shell
    az group create --name AKS-EE-ESA-Demo --location "East US 2"
    az deployment group create \
    --resource-group AKS-EE-ESA-Demo \
    --name akseedemo \
    --template-uri https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_edge_iot_ops_jumpstart/esa_fault_detection/azuredeploy.json \
    --parameters azuredeploy.parameters.json
    ```

>     > **Note:** If you receive an error message stating that the requested VM size is not available in the desired location (as an example: '*Standard_D8s_v3*'), it means that there is currently a capacity restriction for that specific VM size in that particular region. Capacity restrictions can occur due to various reasons, such as high demand or maintenance activities. Microsoft Azure periodically adjusts the available capacity in each region based on usage patterns and resource availability. To continue deploying this scenario, please try to re-run the deployment using another region.

Reviewer: ```suggestion
    > **Note:** If you receive an error message stating that the requested VM size is not available in the desired location (as an example: '*Standard_D8s_v3*'), it could mean that there is currently a capacity restriction for that specific VM size in that particular region. Capacity restrictions can occur due to various reasons, such as high demand or maintenance activities. Microsoft Azure periodically adjusts the available capacity in each region based on usage patterns and resource availability. To continue deploying this scenario, please try to re-run the deployment using another region.
```

![Screenshot Azure Arc-enabled server on resource group](./22.png)

- You can also run _kubectl get nodes -o wide_ to check the cluster node status and _kubectl get pod -A_ to see that the cluster is running and all the needed pods (system and [the Arc-enabled Kubernetes extension pods](https://learn.microsoft.com/azure/azure-arc/kubernetes/extensions)) are in a running state.

> ![Screenshot kubectl get nodes -o wide](./36.png)

Reviewer: screenshot numbers are jumping around and not sequential. Can you please fix the numbers so they are aligned? I know its an annoying thing but it can be confusing if people are just viewing the screenshot files outside of the readme.

![Screenshot kubectl get pod -A](./37.png)

## Edge Storage Accelerator: Real-Time Defect Detection in Manufacturing

Once installed, you will need to open a PowerShell on the *ESA-Win-Demo* server and obtain the IP and service port for the demonstration application. 

![Screenshot kubectl get svc](./32.png)

Use the *esa-webserver-svc* EXTERNAL-IP and PORT to reach the application web frontend.

![Screenshot of web application](./33.png)

Before the application runs, the *Logon.ps1* script will create a container in the *StorageAccountName* the user configured in *azuredeploy.parameters.json*.

![Screenshot of Azure Portal Storage Account](./39.png)

After the application runs, you can validate that the detected objects have been saved to your blob storage account container.

> ![Screenshot of Azure Portal Storage Account](./34.png)

Reviewer: ```suggestion
![Screenshot of Azure portal showing storage account](./34.png)
```

### Exploring logs from the Client VM

Occasionally, you may need to review log output from scripts that run on the _ESA-Win-Demo_ VM in case of deployment failures. To make troubleshooting easier, the scenario deployment scripts collect all relevant logs in the _C:\Temp_ folder on _ESA-Win-Demo_ Azure VM. A short description of the logs and their purpose can be seen in the list below:

| Log file | Description |
| ------- | ----------- |
| _C:\Temp\Bootstrap.log_ | Output from the initial _bootstrapping.ps1_ script that runs on _ESA-Win-Demo_ Azure VM. |

> | _C:\Temp\LogonScript.log_ | Output of _LogonScript.ps1_ which creates the AKS Edge Essentials cluster, onboard it with Azure Arc creating the needed extensions for Edge Storage Accelerator, storage account, storage container, and the kubernetes deployment for running the fault detection scenario |

Reviewer: ```suggestion
| _C:\Temp\LogonScript.log_ | Output of _LogonScript.ps1_ which creates the AKS Edge Essentials cluster, onboards it with Azure Arc and creates the necessary extensions for Edge Storage Accelerator, storage account, storage container, and the Kubernetes deployment for running the fault detection scenario |
```
